event-config.h.cmake

```diff
@@ -474,12 +474,12 @@
 #define EVENT__SIZEOF_VOID_P @EVENT__SIZEOF_VOID_P@
 
 /* set an alias for whatever __func__ __FUNCTION__ is, what sillyness */
-#if defined (__func__)
-#define EVENT____func__ __func__
-#elif defined(__FUNCTION__)
-#define EVENT____func__  __FUNCTION__
-#else
-#define EVENT____func__ __FILE__
+#ifndef __func__
```

Comment:
The `__func__` is not a macro, it is `char[]`, so this will be always triggered while it shouldn't.
For `autotools` we have a `__func__` detection at `configure` step, but for `cmake` we don't.

Reply:
Okay, so what's the workaround?  How do we do similar detection with `cmake` (I'm not versed in `cmake`).